src/framework/core/js/ModelTransformationTransforms.js

```diff
@@ -740,4 +740,159 @@ var fluid = fluid || fluid_2_0_0;
+    /**
+     *
+     * Convert a string to a Boolean, for example, when working with HTML form element values.
+     *
+     * The following are all false: undefined, null, "", "0", "false", false, 0
+     *
+     * Everything else is true.
+     *
+     * @param value {String} The value to be interpreted.
+     * @returns {Boolean} The interpreted value.
+     */
+    fluid.transforms.stringToBoolean = function (value) {
+        if (value) {
+            return !Boolean(value === "" || value === "0" || value === "false");
+        }
+        else {
+            return false;
+        }
+    };
+
+    fluid.defaults("fluid.transforms.stringToBoolean", {
+        gradeNames: ["fluid.standardTransformFunction"]
+    });
+
+    /**
+     *
+     * Convert any value into a stringified boolean, i. e. either "true" or "false".  Anything that evaluates to
+     * true (1, true, "non empty string", {}, et. cetera) returns "true".  Anything else (0, false, null, et. cetera)
+     * returns "false".
+     *
+     * @param value - The value to be converted to a stringified Boolean.
+     * @returns {string} - A stringified boolean representation of the value.
+     */
+    fluid.transforms.booleanToString = function (value) {
+        return value ? "true" : "false";
+    };
+
+    fluid.defaults("fluid.transforms.booleanToString", {
+        gradeNames: ["fluid.standardTransformFunction"]
+    });
+
+    /**
+     *
+     * Transform stringified JSON to an object using `JSON.parse`.  Returns `undefined` if the JSON string is invalid.
+     *
+     * @param value {String} - The stringified JSON to be converted to an object.
+     */
+    fluid.transforms.JSONstringToObject = function (value) {
+        try {
+            return JSON.parse(value);
+        }
+        catch (e) {
+            return undefined;
+        }
+    };
+
+    fluid.defaults("fluid.transforms.JSONstringToObject", {
+        gradeNames: ["fluid.standardTransformFunction"]
+    });
+
+    /**
+     *
+     * Transform an object to a string using `JSON.stringify`.  You can pass the `space` option to be used
+     * as part of your transform, as in:
+     *
+     * ```
+     * "": {
+     *   transform: {
+     *     funcName: "fluid.transforms.objectToJSONString",
+     *     inputPath: "",
+     *     space: 2
+     *   }
+     * }
+     * ```
+     *
+     * The default value for `space` is 0, which disabled spacing and line breaks.
+     *
+     * @param value {Object} - An object to be converted to stringified JSON.
+     *
+     */
+    fluid.transforms.objectToJSONString = function (value, transformSpec) {
+        var space = transformSpec.space || 0;
+        return JSON.stringify(value, null, space);
+    };
+
+    fluid.defaults("fluid.transforms.objectToJSONString", {
+        gradeNames: ["fluid.standardTransformFunction"]
+    });
+
+    /**
+     *
+     * Transform a string to a date using the Date constructor.  Accepts (among other things) the date and dateTime values
```

Comment:
There is a kind of "moral direction" in the framework which suggests that only JSON-serialisable material should be stored in a model. However, I'm loth to impose this as a condition, since there may be important use cases or servings of convenience where users should be allowed to store what they want there, assuming they understand the implications and are happy with the consequences. Right now the consequences are relatively harmless in that the framework (should) treat any non-plain object as immutable. This may change over time when we start to produce more mature authoring environments that expect to serialize everything in a model to disk or network and restore it faithfully on the other side. This time is some distance away, and we may well find ourselves putting schemas on models in order to announce their level of faithfulness, in which case the use of transforms like this could be done on a more formally advertised basis.
You can see the evolution of our thinking on this rather profound issue in the following references: http://docs.fluidproject.org/infusion/development/FrameworkConcepts.html#model-objects which was itself migrated from the rather more laissez-faire and primitive https://wiki.fluidproject.org/display/Infusion13/Model+Objects

So in general I'm happy for these transforms to go in, but I would like a kind of "health warning" comment added both to this comment and to the infusion-docs pull that use of these transforms implies the use of a non-JSON serializable model, which the user does at their own risk, since future versions of the framework may require extra formality in order to allow, or even forbid completely in certain contexts.